hkmc2/shared/src/main/scala/hkmc2/semantics/ImplicitResolver.scala

```diff
@@ -104,20 +105,65 @@ class ImplicitResolver(tl: TraceLogger)
   import ImplicitResolver.ICtx
   import ImplicitResolver.ictx
   
-  def resolve(t: Term)(using ictx: ICtx): Unit = t match
-    case blk: Term.Blk => 
-      resolveBlk(blk)
-    case Apps(Term.TyApp(base, targs), argss) if argss.nonEmpty =>
-      resolveApp(base, S(targs), argss)
-      t.subTerms.foreach(resolve(_))
-    case Apps(base, argss) if argss.nonEmpty =>
-      resolveApp(base, N, argss)
-      t.subTerms.foreach(resolve(_))
-    case _ => 
-      t.subTerms.foreach(resolve(_))
+  enum Expect:
+    case Module(reason: Opt[Message] = N)
```

Comment:
Please, no default args in core logic.

Reply:
Isn't that this default argument fine? The only thing it does is to provide an optional reason, and it does not affect the behavior of the compiler, only the error message produced when the expectation of modulefulness is not met.

Reply:
Error messages are part of the behavior of the compiler!

First, is there any reason to not always include a reason?

Second, the default here buys you exactly one character at the call sites (`N`). Why risk forgetting to pass a reason at a given call site just to save on ONE character?

Reply:
Okay maybe making it explicit is better...

> First, is there any reason to not always include a reason?

Because of the white-list strategy, we are not able to provide reasons for all situations (but of course there is a default message saying something like "unexpected moduleful term ...").